GSA/digital-gov-drupal
# DIGITAL-47 guide nav templates

## Jira ticket




[DIGITAL-42](https://cm-jira.usa.gov/browse/DIGITAL-42)

and 

[DIGITAL-47](https://cm-jira.usa.gov/browse/DIGITAL-47)

## Purpose

Adds content type for guide navigation collection. Adds optional title field for guide nodes to use in navigation collection. Adds paragraph fields for guide nav links and subnav.

New content type: guide nav. It can hold one or more links. Links can optionally have a subnav. Has Checkbox bool fields to match config fields in hugo frontmatter. Adds a reference field so a guide can point to a parent guide to link to above the guide title.

Adds a group of guide nav related fields to the guide content type.
  - A reference to the guidenav the page should use
  - An optional title to use in the guide nav instead of the node's title.

Adds a "guide nav admin" role that can add/edit/delete guidenav nodes.

Adds template and preprocess function to help build:
   - menubar of links to use on guide page
   - next/prev links to display, if enabled
   - display reading time, if enabled
   - sidenav link to display, if enabled



## Includes the following PRs that must be merged first



## Deployment and testing

### Local Setup

`lando si` will re-create the site with a "Test guide nav" node that links to existing guide pages. There's some bug/race condition that doesn't set all the references however. You'll need to manually:

1. Set "web analytics" guide page as the start page for the "test guide nav" node
2. Update each guide paeg and set the "Guide Navigation" ref field to point to "test guide nav"



### QA/Testing instructions



1. menu bar displays, with current link highlighted. "Start page" displays as the "Overview" link
2. Toggling the "Show prev/next" checkbox controls display of those links. Prev/next links shown are correct.
3. Toggling the read time toggle controls the display of the "Estimated Reading Time" div.
4. Toggling the "Show in-page nav" shows the sidenav with expected links
5. Setting a "Parent Guide" ref links the kicker in the header to the selected node.
6. The start page ref is used for the guide header above the guide nav. Same start page displays on all the pages that point back to "Test Guide Nav"
## Checklist for the Developer




## Checklist for the Peer Reviewers
